From 089df070e687748ab09e5cfea20935fc0e1173c3 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Wed, 29 Jun 2022 12:17:57 +0300
Subject: [PATCH] [SW-93688] gaudi2: security: unprotect pcie_wrap vuart at erom_recovery

This patch unprotects pcie_wrap vuart registers when security is
enabled. This is needed for ARC0/1 vuart tx to work when security is
enabled. This change is taken from preboot_rtos repo commit id -
09d4f3db47c5debc7d5b599210a7fdfaf69bb3c4.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I2cc8abe22ba4a6f643279c4ddfdd5ca36e72c6c6
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 718d9b5..6a266e6 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -736,7 +736,7 @@
 
 /* protection bits configurations to open some selective registers */
 static u32 gaudi2_protbits_data0[] = { 0x400, 0xc103 };
-static u32 gaudi2_protbits_data1[] = { 0x40000000 };
+static u32 gaudi2_protbits_data1[] = { 0x3f, 0x40000000 };
 
 struct hl_prot_bits_cfg gaudi2_prot_bits_cfg[] = {
 	{{ HL_BLOCK_TYPE_PSOC, (mmPSOC_GLOBAL_CONF_BASE & ~CFG_BASE), 1, 0, 0, 0x0, 0x0, 0x0 },
@@ -747,7 +747,7 @@
 	},
 	{{ HL_BLOCK_TYPE_PCIE, (mmPCIE_WRAP_BASE & ~CFG_BASE), 1, 0, 0, 0x0, 0x0, 0x0 },
 	  .prot_map = 0x0000000,
-	  .data_map = 0x8000000,
+	  .data_map = 0x8000004,
 	  .data = gaudi2_protbits_data1,
 	  .data_size = ARRAY_SIZE(gaudi2_protbits_data1)
 	},
